lib/Sema/CSDiag.cpp

```diff
@@ -3057,7 +3057,8 @@ bool FailureDiagnosis::diagnoseGeneralConversionFailure(Constraint *constraint){
   // Due to migration reasons, types used to conform to BooleanType, which
   // contain a member var 'boolValue', now does not convert to Bool. This block
   // tries to add a specific diagnosis/fixit to explicitly invoke 'boolValue'.
-  if (toType->isBool()) {
+  if ((fromType->is<MetatypeType>() ||
+       fromType->getRValueType()->getAnyNominal()) && toType->isBool()) {
     auto LookupResult = CS->TC.lookupMember(CS->DC, fromType,
```

Comment:
I thought that it only names sense to lookup `.boolValue` on metatype or nominal, isn't it? And I think it fails in other cases too like in SIL crasher which does `if(import ...`